hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCreateHandle.java

```diff
@@ -49,12 +49,13 @@
   private Path tempPath = null;
   private long recordsWritten = 0;
   private long recordsDeleted = 0;
+  private Iterator<HoodieRecord<T>> recordIterator;
 
   public HoodieCreateHandle(HoodieWriteConfig config, String commitTime, HoodieTable<T> hoodieTable,
-      String partitionPath) {
+      String partitionPath, String fileId) {
```

Comment:
Wanted to do this for a while .. +1. Is this deterministically generated across retries?

Reply:
I just moved new UUID creation to CopyOnWriteLazyInsertIteratable. The retry of this task should ensure a new UUID deterministically. Does that answer your question if I understood it correctly ?

Reply:
@vinothchandar ^

Reply:
yes. thats what I was asking for. thanks.